corehq/apps/reports/standard/forms/reports.py

```diff
@@ -131,12 +132,24 @@ def _fmt_date(somedate):
             else:
                 form_name = EMPTY_FORM
                 form_username = EMPTY_USER
+
+            error_type = SubmissionErrorType.display_name_by_doc_type(xform_dict['doc_type'])
+            if xform_dict['doc_type'] == "XFormArchived":
+                archive_operations = [operation for operation in xform_dict.get('history')
+                                      if operation.get('operation') == 'archive']
+                if archive_operations:
+                    user = CouchUser.get_by_user_id(archive_operations[-1].get('user'))
+                    error_type = "{} {} on {}".format(
+                        user.username if user else "",
```

Comment:
I think you could use `cached_user_id_to_username` here to mitigate some of the performance concerns.

Reply:
ooh, didn't know about this! thanks.